test/jdk/java/net/httpclient/SpecialHeadersTest.java

```diff
@@ -223,42 +382,61 @@ void testHomeMadeIllegalHeader(String uriString,
         out.println("\n--- Starting ");
         final URI uri = URI.create(uriString);
 
-        HttpClient client = HttpClient.newBuilder()
-                .proxy(NO_PROXY)
-                .sslContext(sslContext)
-                .build();
+        HttpClient client = newHttpClient("testHomeMadeIllegalHeader", sameClient);
```

Comment:
Previously before this change, this test method wasn't using the `sameClient` param while creating the new client and instead was creating it always. I don't know if that was intentional (in fact it wasn't even using the `headerNameAndValue` param being passed to it). Is this change to use the `sameClient` param while creating this client here, intentional?

Reply:
yes - before that the method would be running twice with a brand new client. I believe that was an oversight. I don't think that sameClient=true|false matters much here, which probably explains why the sameClient parameters was ignored. We could create a special provider that doesn't pass sameClient, but since this method should be quite short I decided to simply change it to take the value of `sameClient` into account.